File: ems-backend/src/main/java/com/example/emsbackend/service/EmployeeService.java
package com.example.emsbackend.service;

import com.example.emsbackend.dto.EmployeeDto;
import org.springframework.stereotype.Service;

import java.util.List;

public interface EmployeeService
    {
    // Create employee method
    EmployeeDto createEmployee(EmployeeDto employeeDto);
    // Get employee by id method
    EmployeeDto getEmployeeById(Long id);
    // Get all employees method
    List<EmployeeDto> getAllEmployees();
    // Update employee method
    EmployeeDto updateEmployee(Long id, EmployeeDto employeeDto);
    // Delete employee method
    void deleteEmployee(Long id);
    }

File: ems-frontend/src/Services/EmployeeService.js
import axios from "axios";

const REST_API_BASE_URL = "http://localhost:8080/api/employees";

// Get employee by id GET
export const getEmployee = (employeeId) =>
  axios.get(REST_API_BASE_URL + "/" + employeeId);
// Get all employees GET
export const listEmployees = () => axios.get(REST_API_BASE_URL);
// Create employee POST
export const createEmployee = (employee) =>
  axios.post(REST_API_BASE_URL, employee);
// Update employee PUT
export const updateEmployee = (employeeId, employee) =>
  axios.put(REST_API_BASE_URL + "/" + employeeId, employee);
// Delete employee DELETE
export const deleteEmployee = (employeeId) =>
  axios.delete(REST_API_BASE_URL + "/" + employeeId);
